### ISSUE TITLE
The depth/ distance in Mujoco

### ISSUE THREAD

**Issue by cidxb** [deepmind/mujoco#743]
Hi.
So I have use the Depth rendering in Mujoco.
and learn how to get the true distance from the depth image by reading these two links: 👍 
 https://github.com/openai/mujoco-py/issues/520
https://github.com/deepmind/dm_control/blob/409a0579c7f874f16d74ebfd7ef4cb5c959881b8/dm_control/mujoco/engine.py#L734

I have used the World 2 Pixel method to get an object's location in an image(checked and accurate), and I use the Pixel pair(x,y) to read my depth image from the same camera, and read out the distance information processed by  image = near / (1 - self._depth_buffer * (1 - near / far))

I have also read out the camera and the object xpos, using numpy.linalg.norm() to calculate.

But the distance read out in object position (x,y) is 0.056889012
While my calculation using xpos is 0.5249967777189329， which is more likely right.

Why is that near / (1 - self._depth_buffer * (1 - near / far))  result so far wrong?

Thanks for advance !

**Comment by yuvaltassa** [comment]
Why aren't you using the depth image functionality already provided?
This is described in our [tutorial](https://github.com/deepmind/mujoco#getting-started).
Note, this does **not** computed the distance from the position of the camera, but rather the z-depth, i.e. the distance from the plane of the camera.

**Comment by cidxb** [comment]
> Why aren't you using the depth image functionality already provided? This is described in our [tutorial](https://github.com/deepmind/mujoco#getting-started). Note, this does **not** computed the distance from the position of the camera, but rather the z-depth, i.e. the distance from the plane of the camera.

THX. but I did used the rendering method in the tutorial.
The depth image value is not the actual distance value. I want the actual distance...

**Comment by yuvaltassa** [comment]
[Here](https://github.com/deepmind/dm_control/blob/774f46182140106e22725914aad3c6299ed91edd/dm_control/mujoco/engine_test.py#L66) is a _passing test_ of the same code, showing that the distance is correct.

Perhaps this will help you debug?

**Comment by cidxb** [comment]
> [Here](https://github.com/deepmind/dm_control/blob/774f46182140106e22725914aad3c6299ed91edd/dm_control/mujoco/engine_test.py#L66) is a _passing test_ of the same code, showing that the distance is correct.
> 
> Perhaps this will help you debug?

Thank you!  I forget to show my progress.
After reviewing the code here and the instruction in the updated tutorial, I solved my problem.

There are two mistakes I made :
1. The depth image value is natively the depth value, and doesn't need to be processed (with that function  i`mage = near / (1 - self._depth_buffer * (1 - near / far))`)
2. There is a resize value of depth image operation in the tutorial, I check the tutorial a while before, and at that time there isn't a notation, and I thought that resize operation is a standard operation to get real value. Turns out it is just an example for a customized process of the depth image.
THX again!
